### PR TITLE
Add `fetch_distribution_history` and `list_distribution_links`

### PR DESCRIPTION
**Pull request by dsen6644**
Items still left to do

- [ ] Create a warning/message specific to the fetch_distribution_history function that informs those with basic licenses that this function only works for XM directory users. Basic license users will receive a 404 error when calling the function.
- [ ] Tests need cached .yml file in fixtures.

Also noting, I did not use @chrisumphlett error handling. I think that does need to be addressed at a higher level, especially since it's being seen across multiple functions.

Addresses to #206 and #219.

### CODE REVIEW

**Comment by chrisumphlett** [comment]
FWIW when I ran my fork most recently, with the error handling, I never noticed it actually having errors (I didn't look back through the entire console, it was taking 500-700 iterations). When I was using this endpoint a few months ago I was hitting an error every time when doing 100-200 iterations

**Comment by juliasilge** [comment]
So I am having a hard time testing this PR out 😔 because I am not finding out how to "fake" or other mock up distribution data, and I don't have any in my test account. This also means I can't even generate the 404 to test against. I think we are going to need some external input or help to finish this PR up.

What do we need?

- [ ] Someone with a basic license who has used distributions to test _trying_ to get the history and failing so we can generate a nice error
- [ ] Someone with the fancy license who has used distributions who can help get the tests written

Is this right? Anything else that we know of?

File: R/fetch_distribution_history.R [hunk 1]
```diff
@@ -15,7 +16,10 @@
 #'
 #' surveys <- all_surveys()
 #' distributions <- fetch_distributions(surveys$id[1])
-#' distribution_history <- fetch_distribution_history(distributions$id[1])
+#' distribution_history <- fetch_distribution_history(
+#'     distributions$id[1],
+#'     surveyID = surveys$id[1]
+#' )
```
Comment (juliasilge) [inline R/fetch_distribution_history.R:22]: Is this correct? The `surveyID` arg is required, right? (I will fix them to match style-wise in the next commit, with both not needing names.)

Comment (dsen6644) [reply R/fetch_distribution_history.R:22]: Correct, surveyID is required if you want emails associated with the distribution list and not just contact ids. The survey id is necessary for `list_distribution_links` embedded in `fetch_distribution_history`. Feel free to make any style changes you see necessary.

@chrisumphlett may be able to help with the formal testing given that he has an XM directory license.

**Comment by juliasilge** [comment]
Closed in #239

Thank you so much @dsen6644! I attributed your work here in the NEWS 🙌

**Pull request closed without merging**